app/views/devise/shared/_error_messages.html.erb

```diff
@@ -1,15 +1,5 @@
 <% if resource.errors.any? %>
-  <div id="error_explanation">
-    <h2>
-      <%= I18n.t("errors.messages.not_saved",
-                 count: resource.errors.count,
-                 resource: resource.class.model_name.human.downcase)
-       %>
-    </h2>
-    <ul>
-      <% resource.errors.full_messages.each do |message| %>
-        <li><%= message %></li>
-      <% end %>
-    </ul>
-  </div>
+  <% resource.errors.full_messages.each do |message| %>
+    <%= javascript_tag "M.toast({html: '<%= j message =>'}, 4000)" %>
```

Comment:
There is no longer an injection vulnerability, but now the message doesn't show up.  We need the old string interpolation but manual html escape with `#{j(h(message))}`.